### PR TITLE
A field for occurrences of a segment in pan-genome graphs

### LINKED ISSUE #7: Pan-Genome Graphs
There is some recent work on using of de Bruijn graphs on analysis of pan-genome, e.g. see:
1. http://biorxiv.org/content/early/2014/04/06/003954
2. http://link.springer.com/chapter/10.1007/978-3-319-19929-0_4
3. http://link.springer.com/chapter/10.1007/978-3-642-40453-5_17#page-1

So far, there is no common format for such graphs. Does it make sense to generalize GFA to handle both pan-genome/assembly graphs?

### PR DESCRIPTION
**Pull request by ilyaminkin**
Resolves #7

### CODE REVIEW

File: GFA-spec.md [hunk 1]
```diff
@@ -88,6 +88,43 @@ The Sequence field can be `*` meaning that the sequence is not stored in the GFA
...
+	"type": "array",
+	"items": {
+		"title": "Occcurence",
+		"type": "object",
+	       	"properties": {
+			"chr": {
+				"description": "An identifier of the chromosome containing the occurrence",
+				"type": "string"
+			},
+       			"source": {
```
Comment (sjackman) [inline GFA-spec.md]: There is a mixture of spaces and tabs on this line. Please fix.

**Comment by sjackman** [comment]
I'd like to rename `chr` to something more generic, because not all FASTA entries represent chromosomes (could be contigs or scaffolds for example). I suggest `seqid` like GFF.

**Comment by sjackman** [comment]
For `strand`, I've always used `false` for positive strand and `true` for negative strand (and called this field `rc` for reverse-complement, though I prefer the name `strand`), primarily so that the positive entries sort before the negative entries (because 0 sorts before 1 and "False" before "True"). Are there other formats that use a `bool` to represent the strand? With an arbitrary decision like this, it's nice to be consistent with another format. Actually, SAM/BAM in the flags field uses 0 for positive and 1 for negative strand. So, that's my vote.

**Comment by ilyaminkin** [comment]
> I'd like to rename chr to something more generic, because not all FASTA entries represent chromosomes (could be contigs or scaffolds for example). I suggest seqid like GFF.

I agree.

> For strand, I've always used false for positive strand and true for negative strand (and called this field rc for reverse-complement, though I prefer the name strand), primarily so that the positive entries sort before the negative entries (because 0 sorts before 1 and "False" before "True"). Are there other formats that use a bool to represent the strand? With an arbitrary decision like this, it's nice to be consistent with another format. Actually, SAM/BAM in the flags field uses 0 for positive and 1 for negative strand. So, that's my vote.

Probably the most natural way is to use a String with '+' for positive strand and '-' for the negative one, like MAF does: https://cgwb.nci.nih.gov/FAQ/FAQformat.html#format5

File: GFA-spec.md (hunk 1 above)
Comment (ilyaminkin) [inline GFA-spec.md]: Thx for noticing that.

**Comment by ilyaminkin** [comment]
But honestly, I think that the most natural way is to add a separate line for occurrences of a segment (https://github.com/pmelsted/GFA-spec/issues/7#issuecomment-221090503), it is more consistent with the "record"-based idea of the format and simplifies implementation. I realized this after creating this pull request. What do you think?

**Pull request closed without merging**